### PR TITLE
Quantum: Add 'az quantum offerings' commands for provider settings

### PR DESCRIPTION
**Pull request by ricardo-espinoza**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [X] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [X] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [N/A] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
quantum

File: src/quantum/azext_quantum/_params.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ def load_arguments(self, _):
     max_poll_wait_secs_type = CLIArgumentType(help='Poll time in seconds to query Azure Quantum for results of the corresponding job.')
     tag_type = CLIArgumentType(help='Show only quantum workspaces that have associated the specified tag.')
     skip_role_assignment_type = CLIArgumentType(help='Skips the role assignment step for the quantum workspace in the storage account.')
+    provider_id_type = CLIArgumentType(options_list=['--provider-id', '-p'], help='Identifier of an Azure Quantum provider.')
+    sku_type = CLIArgumentType(options_list=['--sku', '-k'], help='Identifies a plan or SKU offered by an Azure Quantum provider.')
```
Comment (qwordy) [inline src/quantum/azext_quantum/_params.py:25]:
```suggestion
    sku_type = CLIArgumentType(options_list=['--sku', '-k'], help='Identify a plan or SKU offered by an Azure Quantum provider.')
```

File: src/quantum/azext_quantum/_params.py [hunk 2]
```diff
@@ -21,6 +21,8 @@ def load_arguments(self, _):
     max_poll_wait_secs_type = CLIArgumentType(help='Poll time in seconds to query Azure Quantum for results of the corresponding job.')
     tag_type = CLIArgumentType(help='Show only quantum workspaces that have associated the specified tag.')
     skip_role_assignment_type = CLIArgumentType(help='Skips the role assignment step for the quantum workspace in the storage account.')
```
Comment (qwordy) [inline src/quantum/azext_quantum/_params.py:23]:
```suggestion
    skip_role_assignment_type = CLIArgumentType(help='Skip the role assignment step for the quantum workspace in the storage account.')
```

File: src/quantum/azext_quantum/operations/offerings.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            publisher_id = p.properties.managed_application.publisher_id
+            break
+    return (publisher_id, offer_id)
+
+
+def _valid_publisher_and_offer(provider, publisher, offer):
+    if (offer is None or publisher is None):
+        raise CLIError(f"Provider '{provider}' not found.")
+    if (offer == OFFER_NOT_AVAILABLE or publisher == PUBLISHER_NOT_AVAILABLE):
+        _show_info(f"No terms exist for provider '{provider}'.")
```
Comment (qwordy) [inline src/quantum/azext_quantum/operations/offerings.py:37]: Is it an error? We seldom use raw `print`.

File: src/quantum/azext_quantum/operations/offerings.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    Accept the terms of a provider and SKU combination to enable it for workspace creation.
+    """
+    client = cf_offerings(cmd.cli_ctx)
+    (publisher_id, offer_id) = _get_publisher_and_offer_from_provider_id(client.list(location_name=location), provider_id)
+    if not _valid_publisher_and_offer(provider_id, publisher_id, offer_id):
+        return None
+    terms_client = cf_vm_image_term(cmd.cli_ctx)
+    terms = terms_client.get(publisher_id, offer_id, sku)
+    terms.accepted = True
+    return terms_client.create(publisher_id, offer_id, sku, terms)
```
Comment (qwordy) [inline src/quantum/azext_quantum/operations/offerings.py:74]: These commands manage vm image terms. It is not necessary to implement again.
https://docs.microsoft.com/en-us/cli/azure/vm/image/terms?view=azure-cli-latest

File: src/quantum/azext_quantum/operations/offerings.py (hunk 3 above)
Comment (ricardo-espinoza) [reply src/quantum/azext_quantum/operations/offerings.py:37]: In this case it's not an error. We will end up returning None from the command, but the message is just a way to let the user know that this is not an error, simply there are no terms to accept for this provider.

File: src/quantum/azext_quantum/operations/offerings.py (hunk 4 above)
Comment (ricardo-espinoza) [reply src/quantum/azext_quantum/operations/offerings.py:74]: Actually, the intent of these commands is precisely to wrap 'az vm image' commands, but doing a translation of provider/sku (which is what Azure Quantum users are acquainted with) and figure out from them the publisherId/Offer/Plan that the 'az vm image' command expects. Also, this level of indirection allows us to extend them if needed to do other things in the quantum workspace in addition to calling the wrapped command.
I'll add a clarifying comment for this.

File: src/quantum/azext_quantum/operations/offerings.py (hunk 3 above)
Comment (ricardo-espinoza) [reply src/quantum/azext_quantum/operations/offerings.py:37]: I'll add a clarifying comment and improve the message of the print output.

File: src/quantum/azext_quantum/operations/offerings.py (hunk 4 above)
Comment (qwordy) [reply src/quantum/azext_quantum/operations/offerings.py:77]: Got it. Thank you for the clear explanation!

File: src/quantum/azext_quantum/operations/offerings.py (hunk 3 above)
Comment (qwordy) [reply src/quantum/azext_quantum/operations/offerings.py]: OK. It is like logger.info or logger.warning. But you want to customize color. No problem.

**Review by qwordy — APPROVED** [review]

**Pull request merged**